bls/signature.go

```diff
@@ -0,0 +1,243 @@
+//go:build relic
+
+package bls
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	flowCrypto "github.com/onflow/flow-go/crypto"
+	"strings"
+)
+
+const SigningAlgorithm = flowCrypto.BLSBLS12381
+
+// TODO: what should the domainTag param be?
+var HashingAlgorithm = flowCrypto.NewExpandMsgXOFKMAC128("deso-protocol")
+
+func AggregateSignatures(signatures []*Signature) (*Signature, error) {
+	var flowSignatures []flowCrypto.Signature
+	for _, signature := range signatures {
+		flowSignatures = append(flowSignatures, signature.flowSignature)
+	}
+	aggregateFlowSignature, err := flowCrypto.AggregateBLSSignatures(flowSignatures)
+	if err != nil {
+		return nil, err
+	}
+	return &Signature{flowSignature: aggregateFlowSignature}, nil
+}
+
+func VerifyAggregateSignature(publicKeys []*PublicKey, signature *Signature, payloadBytes []byte) (bool, error) {
+	var flowPublicKeys []flowCrypto.PublicKey
+	for _, publicKey := range publicKeys {
+		flowPublicKeys = append(flowPublicKeys, publicKey.flowPublicKey)
+	}
+	return flowCrypto.VerifyBLSSignatureOneMessage(flowPublicKeys, signature.flowSignature, payloadBytes, HashingAlgorithm)
+}
+
+//
+// TYPES: PrivateKey
+//
+
+type PrivateKey struct {
```

Comment:
Why did we make all of these structs as opposed to doing something like `type PrivateKey floCrypto.PrivateKey` or something? I think it's fine to leave it as-is, just wondering.

Also Maybe we should call these BLSPrivateKey, BLSPublicKey, BLSSignature, etc...? Up to you. Leave comments either way explaining everything.